Controller/CRUDController.php

```diff
@@ -1364,4 +1364,25 @@ protected function preShow(Request $request, $object)
     protected function preList(Request $request)
     {
     }
+
+    /**
+     * translate a message id.
+     *
+     * @param string $id
+     * @param array  $parameters
+     * @param null   $domain
+     * @param null   $locale
+     *
+     * @return string the translated string
+     */
+    final protected function trans($id, array $parameters = array(), $domain = null, $locale = null)
+    {
+        $domain = $domain ?: $this->admin->getTranslationDomain();
+
+        if (!$this->has('translator')) {
+            return $id;
+        }
+
+        return $this->get('translator')->trans($id, $parameters, $domain, $locale);
+    }
```

Comment:
That part wouldn't change, excecpt we will check against ``$this->has('admin_translator')``